jib-core/src/main/java/com/google/cloud/tools/jib/cache/Caches.java

```diff
@@ -48,25 +83,41 @@ public Initializer setApplicationCacheDirectory(Path applicationCacheDirectory)
       return this;
     }
 
-    public Caches init() throws CacheMetadataCorruptedException, NotDirectoryException {
-      if (baseCacheDirectory == null || applicationCacheDirectory == null) {
-        throw new IllegalStateException(
-            "Must initialize cache with both base image layer cache directory and application image layer cache directory");
+    public Caches init()
+        throws CacheMetadataCorruptedException, IOException, CacheDirectoryNotOwnedException {
+      if (DEFAULT_BASE_CACHE_DIRECTORY.equals(baseCacheDirectory)) {
+        ensureOwnership(DEFAULT_BASE_CACHE_DIRECTORY);
       }
+
       return new Caches(baseCacheDirectory, applicationCacheDirectory);
     }
   }
 
-  public static Initializer initializer() {
-    return new Initializer();
+  /**
+   * @param applicationCacheDirectory Cache for the application image layers - should be local to
+   *     the application project
+   * @return a new {@link Initializer} to initialize the caches.
+   */
+  public static Initializer newInitializer(Path applicationCacheDirectory) {
+    return new Initializer(applicationCacheDirectory);
```

Comment:
I think we need `null` check here or in constructor?

Reply:
I've been using a convention of if the parameter is not annotated with `@Nullable` then it must be non-null to void writing the `@NonNull` annotation. Do you think this is a good convention?

Reply:
oh sure, does error prone catch this? I was thinking we would need to do a preconditions.notNull?

Reply:
Filed https://github.com/google/jib/pull/156